datalad/metadata/__init__.py

```diff
@@ -0,0 +1,333 @@
+      optional auto-detection yielded no results
+    """
+    cfg = ds.config
+    if cfg and cfg.has_section('metadata'):
+        if cfg.has_option('metadata', 'nativetype'):
+            return cfg.get_value('metadata', 'nativetype').split()
+    mtypes = []
+    if guess:
+        # keep local, who knows what some parsers might pull in
+        from . import parsers
+        for mtype in sorted([p for p in parsers.__dict__ if not (p.startswith('_') or p == 'tests')]):
+            pmod = import_module('.%s' % (mtype,), package=parsers.__package__)
+            if pmod.has_metadata(ds):
+                mtypes.append(mtype)
+    return mtypes if len(mtypes) else None
+
+
+# XXX Could become dataset method
+def get_dataset_identifier(ds):
+    """Returns some appropriate identifier for a dataset.
+
+    Any non-annex UUID identifier is prefixed for '_:'
+    """
+    dsid = None
+    if ds.repo:
+        dsid = ds.repo.repo.config_reader().get_value(
+            'annex', 'uuid', default='')
+        if not dsid:
+            # not an annex
+            dsid = '_:{}'.format(ds.repo.get_hexsha())
+    else:
+        # not even a VCS
+        dsid = '_:{}'.format(ds.path.replace(os.sep, '_'))
+
+    return dsid
+
+
+def _get_base_dataset_metadata(ds_identifier):
+    """Return base metadata as dict for a given ds_identifier
+    """
+
+    return {
+        "@context": "http://schema.org/",
+        "@id": ds_identifier,
+        "type": "Dataset",
+    }
+
+
+def get_implicit_metadata(ds, ds_identifier=None):
+    """Convert git/git-annex info into metadata
+
+    Anything that doesn't come as metadata in dataset **content**, but is
+    encoded in the dataset repository itself.
+
+    Returns
+    -------
+    dict
+    """
+    if ds_identifier is None:
+        ds_identifier = get_dataset_identifier(ds)
+
+    meta = _get_base_dataset_metadata(ds_identifier)
+
+    if ds.is_installed():
+        meta['location'] = os.curdir
+
+    # look for known remote annexes, doesn't need configured
+    # remote to be meaningful
+    # need annex repo instance
+    # TODO refactor to use ds.uuid when #701 is addressed
+    if ds.repo and hasattr(ds.repo, 'repo_info'):
+        ds_uuid = ds.repo.repo.config_reader().get_value(
+            'annex', 'uuid', default='')
+        # retrieve possibly stored origin uuid and list in meta data
+        origin_uuid = ''
+        if ds.config:
+            origin_uuid = ds.config.get_value(
+                'annex', 'origin', default='')
+            if origin_uuid and ds_uuid != origin_uuid:
+                meta['dcterms:isVersionOf'] = {'@id': origin_uuid}
+
+        # get all other annex ids, and filter out this one, origin and
+        # non-specific remotes
+        with swallow_logs():
+            # swallow logs, because git annex complains about every remote
+            # for which no UUID is configured -- many special remotes...
+            repo_info = ds.repo.repo_info()
```

Comment:
Moreover it might even try to login to remotes to get their uuid... Yet to see if we should deal with that somehow